contracts/DomainRegistar.sol

```diff
@@ -59,31 +49,48 @@ contract DomainRegistar is Initializable {
      */
     event PriceChanged(uint newPrice, uint oldPrice);
 
+    using DomainUtils for DomainUtils.DomainEntry;
+
     function initialize(uint _domainPrice) public initializer {
         MainStorage storage $ = _getMainStorage();
         $.rootEntry.owner = payable(msg.sender);
         $.rootEntry.weiDomainPrice = _domainPrice;
     }
 
+    function reinitialize() public reinitializer(2) {
+        MainStorage storage $ = _getMainStorage();
+        $.shares[$.rootEntry.owner] = address(this).balance;
+    }
+
+    /**
+     * Return an address of a contract owner
+     */
+    function owner() external view returns (address) {
```

Comment:
line 68 
it is easier to use oppenzeppelin`s Ownable contract upgradeable
// - 1 point

Reply:
in my case usage of Ownable would introduce one extra state variable. I avoid this by keeping contract owner address in root DomainEntry because in fact a contract creator is also an owner of an implicit root domain entry other top-level domains are branched from